docs/outlook/spam-reporting.md

```diff
@@ -0,0 +1,234 @@
+</VersionOverrides>
+```
+
+## Implement the event handler
+
+When your add-in is used to report a message, it generates a [SpamReporting](/javascript/api/office/office.eventtype?view=outlook-js-preview&preserve-view=true) event, which is then processed by the event handler in the JavaScript file of your add-in. To map the name of the event handler you specified in the **\<FunctionName\>** element of your manifest to its JavaScript counterpart, you must call [Office.actions.associate](/javascript/api/office/office.actions#office-office-actions-associate-member) in your code.
+
+Your event handler is responsible for processing the reported message, such as forwarding a copy of the message to an internal system for further investigation. To efficiently send a copy of the reported message, call the [getAsFileAsync](/javascript/api/outlook/office.messageread?view=outlook-js-preview&preserve-view=true#outlook-office-messageread-getasfileasync-member(1)) method in your event handler. This gets the Base64 encoding of a message, which you can then forward to your internal system.
+
+> [!IMPORTANT]
+> To test the `getAsFileAsync` method while it's still in preview in Outlook on Windows, you must configure your computer's registry.
+>
+> Outlook on Windows includes a local copy of the production and beta versions of Office.js instead of loading from the content delivery network (CDN). By default, the local production copy of the API is referenced. To reference the local beta copy of the API, you must configure your computer's registry as follows:
+>
+> 1. In the registry, navigate to `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Office\16.0\Outlook\Options\WebExt\Developer`. If the key doesn't exist, create it.
+> 1. Create an entry named `EnableBetaAPIsInJavaScript` and set its value to `1`.
+>
+>    :::image type="content" source="../images/outlook-beta-registry-key.png" alt-text="The EnableBetaAPIsInJavaScript registry value is set to 1.":::
+
+Once the event handler has completed processing the message, it must call the [event.completed](/javascript/api/office/office.addincommands.event#office-office-addincommands-event-completed-member(1)) method. In addition to signaling to the add-in that the spam-reporting event has been processed, `event.completed` can also be used to customize a post-processing dialog to show to the user or perform additional operations on the message, such as deleting it from the inbox. For a list of properties you can include in a JSON object to pass as a parameter to the `event.completed` method, see [Office.AddinCommands.EventCompletedOptions](/javascript/api/office/office.addincommands.eventcompletedoptions).
+
+> [!NOTE]
+> Code added after the `event.completed` call isn't guaranteed to run.
+
+The following is an example of a spam-reporting event handler.
+
+```javascript
+// Handles the SpamReporting event to process a reported message.
+function onSpamReport(event) {
+  // Gets the Base64 encoding of a reported message.
```

Comment:
we should mention that typically a copy of the message is required so this is what the snippet is showing.

Reply:
Clarified this in the paragraphs before the snippet.